addons/html_builder/static/src/website_builder/plugins/options/google_map_option/google_map_option.js

```diff
@@ -21,12 +21,15 @@ export class GoogleMapOption extends BaseOptionComponent {
         /** @type {Props} */
         this.props;
         this.inputRef = useRef("inputRef");
-        /** @type {{ map: Element }} */
-        this.state = useDomState(map => ({ map }));
-        onWillStart(async () => {
-            await this.props.loadGoogleMaps(this.state.map);
-        });
-        onMounted(() => {
+        /** @type {{ map: Element, formattedAddress: string }} */
+        this.state = useDomState(map => ({
+            map,
+            formattedAddress: map.dataset.pinAddress || "",
+        }));
+        useEffect(() => {
+            this.state.map.dataset.pinAddress = this.state.formattedAddress;
+        }, () => [ this.state.formattedAddress ]);
```

Comment:
You should never use useEffect to edit the state. It will add a flicker. You trigger a second animation frame.
You should probably use effect in your case.
Or maybe not edit the state. maybe map shouldn t be in the state ? It s just the current editingElement (this.env.getEditingElement()